src/main/scala/org/allenai/deep_qa/data/SquadTaggingDatasetReader.scala

```diff
@@ -0,0 +1,108 @@
+ * using a parser and output a pre-tokenized token-tagged Instance.
+ */
+class SquadTaggingDatasetReader(
+  params: JValue
+) extends DatasetReader[BaseTaggingInstance] {
+  val tokenize = JsonHelper.extractWithDefault(params, "tokenize", true)
+
+  lazy val fileUtil = new FileUtil
+  lazy val parser = new StanfordParser()
+  override def readFile(filename: String): Dataset[BaseTaggingInstance] = {
+    val json = parse(fileUtil.readFileContents(filename))
+    val instanceTuples = for {
+      JObject(article) <- json \ "data"
+      JField("paragraphs", JArray(paragraphs)) <- article
+      JObject(paragraph_jval) <- paragraphs
+      JField("context", JString(paragraph)) <- paragraph_jval
+      JField("qas", JArray(questions)) <- paragraph_jval
+    } yield (paragraph.replace("\n", " "), questions)
+
+    val instances = instanceTuples.par.map { case (paragraph, questions) => {
+      val answerTuples = for {
+        JObject(question_jval) <- questions
+        JField("question", JString(question)) <- question_jval
+        JField("answers", JArray(answers)) <- question_jval
+        JObject(answer_jval) <- answers
+        JField("answer_start", JInt(answerStart)) <- answer_jval
+        JField("text", JString(answerText)) <- answer_jval
+      } yield (answerStart.toInt, answerText)
+      val uniqueAnswers = answerTuples.groupBy(identity).mapValues(_.size).toList.sortBy(-_._2).map(_._1)
+      val keptAnswers = resolveConflicts(uniqueAnswers)
+      if (!tokenize) {
+        val spanIndices = keptAnswers.map { case (spanStart, spanText) => { (spanStart, spanStart + spanText.size) }}
+        SequenceTaggingInstance(paragraph, Some(Map("answers" -> spanIndices).toString))
+      } else {
+        tagPassage(paragraph, keptAnswers)
+      }
+    }}
+    Dataset(instances.seq)
+  }
+
+  /**
+   * Takes a sequence of spans from a passage, specified as (start position, text), and removes
+   * overlaps.  If there's a subset relationship, we keep the longer one.  If there's a partial
+   * overlap, we pick semi-randomly (whichever one is longer).
+   */
+  def resolveConflicts(spans: Seq[(Int, String)]): Seq[(Int, String)] = {
+    // Biggest spans first; then we'll add them to a set and just ignore later spans that fully or
+    // partially overlap with already-kept spans.
+    val sortedSpans = spans.sortBy(-_._2.size)
+    val keptSpans = new mutable.HashSet[(Int, String)]
+    for (candidateSpan <- sortedSpans) {
+      val candidateSpanStart = candidateSpan._1
+      val candidateSpanEnd = candidateSpanStart + candidateSpan._2.size
+      val overlaps = keptSpans.map { case (spanStart, spanText) => {
+        val spanEnd = spanStart + spanText.size
+        val beginOverlaps = candidateSpanStart >= spanStart && candidateSpanStart <= spanEnd
+        val endOverlaps = candidateSpanEnd >= spanStart && candidateSpanStart <= spanEnd
+        beginOverlaps || endOverlaps
+      }}
+      if (!overlaps.foldLeft(false)(_ || _)) {
```

Comment:
Ok I just discussed this with Marc because I was interested what he thought and he convinced me that your way is better, because the logic is explicit, whereas with `exists` it is implicitly `and`. I retract my comment 😄